monitoring/monitoring_on.go

```diff
@@ -0,0 +1,46 @@
+// +build monitoring
+
+package monitoring
+
+import (
+	// "fmt"
+	"net/http"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Enabled signifies whether the monitoring tag was specified when building lnd
```

Comment:
I think this can be lifted into the config above, so it's possible to build with the tag, but not always have the extra HTTP server listening.